libs/compact-map/src/Data/Compact/KeyMap.hs

```diff
@@ -555,6 +565,23 @@ mapWithKey f (Full arr) = Full (fmap (mapWithKey f) arr)
 instance Functor KeyMap where
   fmap f x = mapWithKey (\_ v -> f v) x
 
+instance Foldable KeyMap where
+  length = size
+  foldr f = foldWithDescKey (const f)
+
```

Comment:
We really should have a strict foldl' functiob

Reply:
Strict `foldl'` is given to us by `Foldable`, this was just a minimal implementation needed. In fact, because `foldWithDescKey` is strict even the lazy `foldr` is currently strict, so what we really need is a lazy foldr

I'd like to create a second PR with optimizations to `SplitMap`. Because it will require additions of bunch of benchmarks, I'd rather do it as a separate PR instead of bloating this one even more

Reply:
Done